synapse/tests/test_lib_cell.py

```diff
@@ -180,202 +183,210 @@ class CellTest(s_t_utils.SynTest):
-            pathinfo = await cell.setDriveInfoPath(baziden, 'zipzop')
-            self.len(1, pathinfo)
-            self.eq(s_drive.rootdir, pathinfo[-1].get('parent'))
+                info = await cell.setDriveInfoPerm(baziden, {'users': {rootuser: 3}, 'roles': {}})
+                self.eq(3, info['perm']['users'][rootuser])
 
-            pathinfo = await cell.setDriveInfoPath(baziden, 'hehe/haha/hoho')
-            self.len(3, pathinfo)
+                with self.raises(s_exc.NoSuchIden):
+                    # s_drive.rootdir is all 00s... ;)
+                    await cell.setDriveInfoPerm(s_drive.rootdir, {'users': {}, 'roles': {}})
+
+                await cell.addDrivePath('hehe/haha')
+                pathinfo = await cell.setDriveInfoPath(baziden, 'hehe/haha/hoho')
+
+                self.eq('hoho', pathinfo[-1].get('name'))
+                self.eq(baziden, pathinfo[-1].get('iden'))
+
+                self.true(await cell.drive.hasPathInfo('hehe/haha/hoho'))
+                self.false(await cell.drive.hasPathInfo('foo/bar/baz'))
+
+                pathinfo = await cell.getDrivePath('foo/bar')
+                self.eq(0, pathinfo[-1].get('kids'))
+
+                pathinfo = await cell.getDrivePath('hehe/haha')
+                self.eq(1, pathinfo[-1].get('kids'))
+
+                with self.raises(s_exc.DupName):
+                    iden = pathinfo[-2].get('iden')
+                    name = pathinfo[-1].get('name')
+                    cell.drive.reqFreeStep(iden, name)
+
+                walks = [item async for item in cell.drive.walkPathInfo('hehe')]
+                self.len(3, walks)
+                # confirm walked paths are yielded depth first...
+                self.eq('hoho', walks[0].get('name'))
+                self.eq('haha', walks[1].get('name'))
+                self.eq('hehe', walks[2].get('name'))
+
+                iden = walks[2].get('iden')
+                walks = [item async for item in cell.drive.walkItemInfo(iden)]
+                self.len(3, walks)
+                self.eq('hoho', walks[0].get('name'))
+                self.eq('haha', walks[1].get('name'))
+                self.eq('hehe', walks[2].get('name'))
+
+                self.none(cell.drive.getTypeSchema('newp'))
+
+                cell.drive.validators.pop('woot')
+                self.nn(cell.drive.getTypeValidator('woot'))
+
+                # move to root dir
+                pathinfo = await cell.setDriveInfoPath(baziden, 'zipzop')
+                self.len(1, pathinfo)
+                self.eq(s_drive.rootdir, pathinfo[-1].get('parent'))
+
+                pathinfo = await cell.setDriveInfoPath(baziden, 'hehe/haha/hoho')
+                self.len(3, pathinfo)
+
+            async with self.getTestCell(dirn=dirn) as cell:
+                data = {'type': 'woot', 'size': 20, 'stuff': 12, 'woot': 'woot'}
+                # explicitly clear out the cache JsValidators, otherwise we get the cached, pre-msgpack
```

Comment:
To avoid mutating globals, we can also confirm that `getTypeSchema` does not return any tuples?

Reply:
Since this is a cache, i think it's ok to just clear it, although i would probably use
```
s_config._JsValidators.clear()
```